hoodie-common/src/main/java/com/uber/hoodie/common/table/HoodieView.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.uber.hoodie.common.table;
+
+import com.uber.hoodie.common.model.CompactionOperation;
+import com.uber.hoodie.common.table.timeline.HoodieInstant;
+import com.uber.hoodie.common.util.Option;
+import com.uber.hoodie.common.util.collection.Pair;
+import java.util.stream.Stream;
+
+/*
+ * A consolidate file-system view interface exposing both realtime and read-optimized views along with
+ * update operations.
+ */
+public interface HoodieView extends TableFileSystemView, TableFileSystemView.ReadOptimizedView,
```

Comment:
we should definitely rename this.  Its very confusing since when we usually refer to Views, its read optimized, real time and incremental views.. This is still a FileSystemView..